src/Aspire.Hosting.SqlServer/SqlServerBuilderExtensions.cs

```diff
@@ -1,17 +1,27 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Globalization;
+using System.Text.RegularExpressions;
+using System.Text;
 using Aspire.Hosting;
 using Aspire.Hosting.ApplicationModel;
+using Microsoft.Data.SqlClient;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 
 namespace Aspire.Hosting;
 
 /// <summary>
 /// Provides extension methods for adding SQL Server resources to the application model.
 /// </summary>
-public static class SqlServerBuilderExtensions
+public static partial class SqlServerBuilderExtensions
 {
+    // GO delimiter format: {spaces?}GO{spaces?}{repeat?}{comment?}
+    // https://learn.microsoft.com/sql/t-sql/language-elements/sql-server-utilities-statements-go
+    [GeneratedRegex(@"^(?:\s*)(GO|go)(?<repeat>\s+\d{1,6})?(?:\s*\-{2,}.*)?(?:\s+)?$", RegexOptions.CultureInvariant)]
```

Comment:
```suggestion
    [GeneratedRegex(@"^\s*(GO|go)(?<repeat>\s+\d{1,6})?\s*(?:-{2,}.*)?\s*$", RegexOptions.CultureInvariant)]
```
i don't think you need the non capturing groups?

Reply:
uh, I didn't test it though

Reply:
```suggestion
    [GeneratedRegex(@"^\s*GO(?<repeat>\s+\d{1,6})?\s*(?:-{2,}.*)?\s*$", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase)]
```
?

Reply:
Thanks, suggestions applied, and added unit tests.